internal/server/healthService.go

```diff
@@ -103,5 +103,47 @@ func (h *HealthService) GetMentalHealthScoreForUser(
 	return res, err
 }
 
+func (h *HealthService) DeleteHealthDataForUser(
+	ctx context.Context,
+	req *pbhealth.DeleteHealthDataForUserRequest,
+) (*pbhealth.DeleteHealthDataForUserResponse, error) {
+	var err error
+	var numDeleted uint32
+
+	switch x := req.Data.(type) {
+	case *pbhealth.DeleteHealthDataForUserRequest_All:
+		numDeleted, err = h.db.DeleteMentalHealthLogs(ctx, req.UserID, nil, x.All)
+		break
+	case *pbhealth.DeleteHealthDataForUserRequest_DateToRemove:
+		numDeleted, err = h.db.DeleteMentalHealthLogs(ctx, req.UserID, x.DateToRemove, true)
+
+	}
+
+	res := &pbhealth.DeleteHealthDataForUserResponse{EntriesDeleted: numDeleted}
+
+	return res, err
+}
+
+func (h *HealthService) UpdateHealthDataForDate(
+	ctx context.Context,
+	req *pbhealth.UpdateHealthDataForDateRequest,
+) (*pbhealth.UpdateHealthDataForDateResponse, error) {
+
+	err := h.db.UpdateMentalHealthLogs(ctx, req.UserID, req.DesiredLogInfo)
+	if err != nil {
+		h.logger.Infof("%v", err)
+		return &pbhealth.UpdateHealthDataForDateResponse{
+			Success: false,
+		}, status.Errorf(codes.Internal, "Error updating mental health logs")
```

Comment:
Probably should use something other than the internal error code here, perhaps invalid arguments or something

Reply:
Changed it to Invalid Argument. Thanks!